src/Windows/R/Components/Impl/Plots/Implementation/Commands/PlotHistoryCommand.cs

```diff
@@ -2,17 +2,50 @@
 // Licensed under the MIT License. See LICENSE in the project root for license information.
 
 using System;
+using System.Linq;
+using System.Threading.Tasks;
 using Microsoft.Common.Core.Diagnostics;
+using Microsoft.Common.Core.Shell;
+using Microsoft.Common.Core.UI.Commands;
 using Microsoft.R.Components.InteractiveWorkflow;
 
 namespace Microsoft.R.Components.Plots.Implementation.Commands {
     internal abstract class PlotHistoryCommand : InteractiveWorkflowAsyncCommand {
         protected IRPlotHistoryVisualComponent VisualComponent { get; }
 
-        public PlotHistoryCommand(IRInteractiveWorkflow interactiveWorkflow, IRPlotHistoryVisualComponent visualComponent) :
+        protected PlotHistoryCommand(IRInteractiveWorkflow interactiveWorkflow, IRPlotHistoryVisualComponent visualComponent) :
             base(interactiveWorkflow) {
             Check.ArgumentNull(nameof(visualComponent), visualComponent);
             VisualComponent = visualComponent;
         }
+
+        public virtual CommandStatus Status {
+            get {
+                var selection = VisualComponent.SelectedPlots.ToArray();
```

Comment:
`PlotHistoryAutoHideCommand`, `PlotHistoryZoomInCommand` and `PlotHistoryZoomOutCommand` have no dependency on selection. I think there should be a separate base class for `PlotHistoryActivateCommand`, `PlotHistoryRemoveCommand` and `PlotHistoryCutCopyCommand`. Something like `PlotHistorySelectionCommand`.